diff --git a/ryu/hooks.py b/ryu/hooks.py
index d7a2a8dd..f9ae002e 100644
--- a/ryu/hooks.py
+++ b/ryu/hooks.py
@@ -17,7 +17,14 @@
 
 import os
 import sys
-from setuptools.command import easy_install
+
+try:
+    from setuptools.command import easy_install
+    _orig_get_script_args = easy_install.get_script_args
+except (ImportError, AttributeError):
+    easy_install = None
+    _orig_get_script_args = None
+
 from ryu import version
 
 
@@ -33,7 +40,8 @@ def save_orig():
     """Save original easy_install.get_script_args.
     This is necessary because pbr's setup_hook is sometimes called
     before ours."""
-    _main_module()._orig_get_script_args = easy_install.get_script_args
+    if _orig_get_script_args is not None:
+        _main_module()._orig_get_script_args = _orig_get_script_args
 
 
 def setup_hook(config):
@@ -53,11 +61,12 @@ def setup_hook(config):
     # but we are in patching wars already...
     from pbr import packaging
 
-    def my_get_script_args(*args, **kwargs):
-        return _main_module()._orig_get_script_args(*args, **kwargs)
+    if _orig_get_script_args is not None:
+        def my_get_script_args(*args, **kwargs):
+            return _main_module()._orig_get_script_args(*args, **kwargs)
 
-    packaging.override_get_script_args = my_get_script_args
-    easy_install.get_script_args = my_get_script_args
+        packaging.override_get_script_args = my_get_script_args
+        easy_install.get_script_args = my_get_script_args
 
     # another hack to allow setup from tarball.
     orig_get_version = packaging.get_version
